datafusion/core/tests/dataframe/mod.rs

```diff
@@ -6479,3 +6479,87 @@ async fn test_duplicate_state_fields_for_dfschema_construct() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn array_distinct_on_list_with_inner_nullability_causing_type_mismatch(
```

Comment:
I don't like this test, I wish I could write `slt`. 
But I don't know how to construct a nullable array from sql.

Maybe with this improvement from `arrow-rs` (https://github.com/apache/arrow-rs/issues/8351), 
we can write something like this in the future:
`select arrow_cast([1,2,3], 'List(Int64)')` (for nullable = false)
and
`select arrow_cast([1,2,3], 'List(nullable Int64)')` (for nullable = true)

(then lots of tests with nested data type can be rewritten into `slt` tests)